src/pages/invite-employee/__tests__/invite-employee.component.test.js

```diff
@@ -0,0 +1,9 @@
+import wrapper from 'test/test-utils';
+import InviteEmployee from '../invite-employee.component';
+
+describe('<InviteEmployee />', () => {
+  it('renders correctly', () => {
+    const rendered = wrapper(InviteEmployee);
+    expect(rendered.container).toMatchSnapshot();
+  });
+});
```

Comment:
I usually don't like to use snapshots to test a page because it has lots of lines of code which makes it hard to review. In this specific test, we don't test the case when the `isLoading` is `true`, for example:

https://user-images.githubusercontent.com/54550926/170495122-ce818837-5a4e-4772-8459-fd6a0bd3e5b1.mp4

If we break the code of the component, the test might not catch errors:

https://user-images.githubusercontent.com/54550926/170495239-b354ecb2-a54a-428c-9c83-dfedb7118352.mp4

Reply:
I don't like snapshoting very much either, but it can be handy when the UI unexpectedly change without us noticing it.
So I just wrote a snapshot test for the only sake of testing and because I had a hard time trying to unit test this component with Jest and RTL. Actually I used Cypress to unit test this page to make sure the UI components are rendered.
 
And you're right I missed testing the case scenario when `isLoading` is `true` and when some other brake change happens. I'll try to implement it.

I appreciate you documenting your point of view with videos!

Reply:
> it can be handy when the UI unexpectedly change without us noticing it.

But I feel like that is another problem. WIth RTL we should try to test based on the user perspective, the users don't have to know if the element is being rendered inside one or two divs, they only care if it is being rendered at all, and that's what we should check.
(Not that I don't like snapshots, I like to use them for small components, when we don't want to change often and any change might break other components.)

> Actually I used Cypress to unit test this page to make sure the UI components are rendered.

Yeah, that's something I noticed, I feel like those tests could be component tests.

> So I just wrote a snapshot test for the only sake of testing and because I had a hard time trying to unit test this component with Jest and RTL.

Yeah, I know how component tests can be difficult, especially using a component library that has more code than we imagine hahaha. So let me share with you some basic unit tests examples, just to try to clarify/help a little bit:

![code](https://user-images.githubusercontent.com/54550926/170550345-469be197-1e7e-45b4-9f65-d8fd5da87844.png)

The only change I made was in lines 53 and 56:

![image](https://user-images.githubusercontent.com/54550926/170550592-181b8b0f-97a1-421a-b7df-5ff56204173d.png)

There are room to add more tests, tbh, but I feel like those are the basic ones

Reply:
Let me know if you have any questions or if you need help with those tests, you can ping me on discord

Reply:
Btw, the test output would be:
![image](https://user-images.githubusercontent.com/54550926/170551217-0555e2c8-f715-46dc-8c1a-df06cd6d5aa9.png)

Reply:
Let me know if you have any questions or if you need some help with those tests, you can ping me on discord